clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -554,6 +554,36 @@ def StructLayoutAttr : CIR_Attr<"StructLayout", "struct_layout"> {
 // DynamicCastInfoAttr
 //===----------------------------------------------------------------------===//
 
+def DynamicCastToVoidInfoAttr
+    : CIR_Attr<"DynamicCastToVoidInfo", "dyn_cast_to_void_info"> {
+  let summary = "ABI specific information about a dynamic cast to void";
+  let description = [{
+    Provide ABI specific information about a dynamic cast operation that casts
+    a struct pointer to a void pointer.
+
+    The `vtableElemTy` gives the type of elements in the vtable of the input
+    object. The `vtableElemAlign` gives the alignment of elements in the vtable
+    of the input object.
+  }];
+
+  let parameters = (ins "mlir::Type":$vtableElemTy,
+                        "uint64_t":$vtableElemAlign);
+
+  let builders = [
+    AttrBuilderWithInferredContext<(ins "mlir::Type":$vtableElemTy,
+                                        "uint64_t":$vtableElemAlign), [{
+      return $_get(vtableElemTy.getContext(), vtableElemTy, vtableElemAlign);
+    }]>,
+  ];
+
+  let assemblyFormat = [{
+    `<`
+      `vtableElemTy` `=` qualified($vtableElemTy) `,`
```

Comment:
As a general approach, names printed in CIR asm should never have this type of case, I found one the other day that I missed during review and remove it, please let's not introduce additional ones. So something like this should be instead `vtable_elt_type`.

Reply:
Updated.